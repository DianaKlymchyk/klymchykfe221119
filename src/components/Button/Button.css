button {
  max-height: 46px;
}