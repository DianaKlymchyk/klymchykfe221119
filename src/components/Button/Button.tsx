import React, {FC, useState} from 'react';
import './Button.css';

interface Props {
    text: string;
    typeOfWidth?: string;
    onClickCallback?: () => void;
}

const STATUS = {
    HOVERED: 'hovered',
    NORMAL: 'normal',
};

const Button: FC<Props> = ({text,typeOfWidth, onClickCallback}) => {
    const [status, setStatus] = useState(STATUS.NORMAL);

    const onMouseEnter = () => {
        setStatus(STATUS.HOVERED);
    };

    const onMouseLeave = () => {
        setStatus(STATUS.NORMAL);
    };

    return (
        <button
            className={`view-button ${status}`}
            style={{width: typeOfWidth === 'full' ? '100%' : 'auto'}}
            onClick={() => onClickCallback && onClickCallback()}
            onMouseEnter={onMouseEnter}
            onMouseLeave={onMouseLeave}
        >
            {text}
        </button>
    );
}

export default Button;
