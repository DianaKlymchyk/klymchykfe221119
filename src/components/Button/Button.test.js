import renderer from 'react-test-renderer';
import Button from "./Button";

it('changes the class when hovered', () => {
    const component = renderer.create(
      <Button>Text</Button>,
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    renderer.act(() => {
        tree.props.onMouseEnter();
    });
    tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    renderer.act(() => {
        tree.props.onMouseLeave();
    });
    tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});