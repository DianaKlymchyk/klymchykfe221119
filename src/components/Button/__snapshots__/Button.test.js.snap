// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`changes the class when hovered 1`] = `
<button
  className="view-button normal"
  onClick={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  style={
    Object {
      "width": "auto",
    }
  }
/>
`;

exports[`changes the class when hovered 2`] = `
<button
  className="view-button hovered"
  onClick={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  style={
    Object {
      "width": "auto",
    }
  }
/>
`;

exports[`changes the class when hovered 3`] = `
<button
  className="view-button normal"
  onClick={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  style={
    Object {
      "width": "auto",
    }
  }
/>
`;
